File: examples/dotnet/Internal/NotificationAgent.cs
using System.Text.Json.Nodes;
using Axme.Sdk;
using AxmeExamples;

var baseUrl = SseHelper.Env("AXME_BASE_URL", "https://api.cloud.axme.ai");
var apiKey = SseHelper.Env("AXME_API_KEY");
var addr = SseHelper.Env("AXME_AGENT_ADDRESS", "risk-assessment-agent");
if (string.IsNullOrEmpty(apiKey)) { Console.Error.WriteLine("AXME_API_KEY is required"); return; }

var client = new AxmeClient(new AxmeClientConfig { ApiKey = apiKey, BaseUrl = baseUrl });
Console.WriteLine($"risk-assessment-agent starting  address={addr}");
var since = 0;

while (true)
{
    try
    {
        foreach (var d in await SseHelper.PollAgentStreamAsync(baseUrl, apiKey, addr, since))
        {
            var id = SseHelper.Str(d, "intent_id");
            if (d["seq"]?.GetValue<int>() is int seq) since = Math.Max(since, seq);
            if (string.IsNullOrEmpty(id)) continue;
            var intent = SseHelper.UnwrapIntent(await client.GetIntentAsync(id));
            if (!SseHelper.IsActionable(intent)) continue;
            var p = SseHelper.EffectivePayload(intent);

            var ver = SseHelper.Str(p, "version", "0.0.0"); var env = SseHelper.Str(p, "environment");
            var cc = (int)(p["change_count"]?.GetValue<double>() ?? 0);
            var thresh = (int)(p["risk_threshold"]?.GetValue<double>() ?? 30);
            var parts = ver.Split('.'); int.TryParse(parts[0], out var major); var minor = parts.Length > 1 && int.TryParse(parts[1], out var m) ? m : 0;

            var score = 0; var factors = new List<string>();
            if (major >= 4) { score += 40; factors.Add($"major bump +40"); }
            else if (minor >= 5) { score += 15; factors.Add("minor bump +15"); }
            if (cc > 30) { score += 30; factors.Add($"high changes +30"); }
            if (env == "production") { score += 25; factors.Add("prod +25"); }
            else if (env is "staging" or "canary") { score += 10; factors.Add("staging +10"); }
            var level = score >= thresh ? "HIGH" : score >= thresh / 2 ? "MEDIUM" : "LOW";

            var res = new JsonObject { ["action"] = "complete", ["risk_score"] = score, ["risk_level"] = level,
                ["factors"] = new JsonArray(factors.Select(f => (JsonNode)JsonValue.Create(f)!).ToArray()) };
            await client.ResumeIntentAsync(id, res, new RequestOptions { OwnerAgent = addr });
            Console.WriteLine($"resumed {id} (risk={level} score={score})");
        }
    }
    catch { await Task.Delay(2000); }
}

File: examples/internal/notification/java/NotificationAgent.java
package ai.axme.examples.internal;

import ai.axme.examples.SseHelper;
import dev.axme.sdk.*;
import java.util.*;

/** Risk Assessment Agent — internal/notification example (Java). */
public class NotificationAgent {
    static final String BASE = SseHelper.env("AXME_BASE_URL", "https://api.cloud.axme.ai");
    static final String KEY = System.getenv("AXME_API_KEY");
    static final String ADDR = SseHelper.env("AXME_AGENT_ADDRESS", "risk-assessment-agent");

    public static void main(String[] args) throws Exception {
        if (KEY == null || KEY.isEmpty()) { System.err.println("AXME_API_KEY is required"); System.exit(1); }
        AxmeClient client = new AxmeClient(new AxmeClientConfig(BASE, KEY));
        System.out.printf("risk-assessment-agent starting  address=%s%n", ADDR);
        int since = 0;
        while (true) {
            try {
                for (Map<String, Object> d : SseHelper.pollAgentStream(BASE, KEY, ADDR, since, 15)) {
                    String id = SseHelper.str(d, "intent_id"); Number seq = (Number) d.get("seq");
                    if (seq != null) since = Math.max(since, seq.intValue()); if (id.isEmpty()) continue;
                    Map<String, Object> intent = SseHelper.unwrapIntent(client.getIntent(id, RequestOptions.none()));
                    if (!SseHelper.isActionable(intent)) continue;
                    Map<String, Object> p = SseHelper.effectivePayload(intent);

                    String ver = SseHelper.str(p, "version", "0.0.0"), env = SseHelper.str(p, "environment");
                    int cc = p.get("change_count") instanceof Number n ? n.intValue() : 0;
                    int thresh = p.get("risk_threshold") instanceof Number n ? n.intValue() : 30;
                    String[] parts = ver.split("\\."); int major = parts.length > 0 ? Integer.parseInt(parts[0]) : 0;
                    int minor = parts.length > 1 ? Integer.parseInt(parts[1]) : 0;

                    int score = 0; List<String> factors = new ArrayList<>();
                    if (major >= 4) { score += 40; factors.add("major bump +40"); }
                    else if (minor >= 5) { score += 15; factors.add("minor bump +15"); }
                    if (cc > 30) { score += 30; factors.add("high changes +30"); }
                    if ("production".equals(env)) { score += 25; factors.add("prod +25"); }
                    else if ("staging".equals(env) || "canary".equals(env)) { score += 10; factors.add("staging +10"); }
                    String level = score >= thresh ? "HIGH" : score >= thresh / 2 ? "MEDIUM" : "LOW";

                    client.resumeIntent(id, Map.of("action", "complete", "risk_score", score, "risk_level", level, "factors", factors),
                            SseHelper.withOwner(ADDR));
                    System.out.printf("resumed %s (risk=%s score=%d)%n", id, level, score);
                }
            } catch (Exception e) { Thread.sleep(2000); }
        }
    }
}
